Javadoc index in reverse order and link to PonyMail

nacx opened the pull request.

jclouds-commentator commented:
Go to http://50d0c57bec39f04aaa32-5733c3472e65647debd31c25780fa066.r14.cf5.rackcdn.com/ to review your changes.

demobox commented on community/index.md:
```diff
@@ -8,6 +8,7 @@ title: Get in touch!
     <p>We'd love to have your feedback, suggestions, and will be happy to help with anything you need. Just get in touch!</p>
 
     <ul>
+      <li><a href="https://lists.apache.org/list.html?jclouds.apache.org">Browse and participate</a> in our mailing lists</li>
```
For me, this link defaults to `dev@`, for some reason (alphabetical order)? Is that what we want?

![image](https://cloud.githubusercontent.com/assets/223702/21071458/bd495f5c-be6e-11e6-8060-093d4509a4a5.png)

demobox replied:
> Makes sense?

The explanation makes sense, but I wonder if that would encourage people to simply use the dev list for everything? In general, I wonder whether "default to user@" would be a better guide in terms of where to ask questions first?

@andreaturli @andrewgaul Would be curious to hear your thoughts on that?

andreaturli replied:
I think "default to user@" makes sense to me as well, but strong feeling though

nacx replied:
I don't have a strong opinion neither (happy to change to `user@`), but let me explain the purpose of this change: encourage participation.

PonyMail allows users to subscribe to specific threads, or even to participate in the discussions without having to subscribe to the lists, which can be "difficult" or inconvenient for users that are just interested in a particular thing. If we want to encourage participation from newcomers, perhaps is better to direct them by default to an active list instead to a very quiet one, so they see there is actually people in there. With the current email rate we could assume emails coming to the dev list by mistake.

I'm happy to default to the user list though, as it makes more sense and it could be the *right list*. However, I think it could be more convenient to default to dev. Dunno! :)

demobox replied:
> let me explain the purpose of this change

Thanks for the useful and interesting explanation - is there any way we can actually include that in the PR in some way, i.e. make it visible to the user somehow?

I feel that what you wrote is information that would be very useful to the users, too; perhaps there is an easy-ish way to explain the purpose of the various lists somewhere in more detail _before_ sending users to this page? Or perhaps there is a view that shows _all_ activity (across lists), so that we don't need to pick one and hope that it's the "right one"?

demobox commented on reference/javadoc/index.md:
```diff
@@ -4,7 +4,7 @@ title: Javadoc for Apache jclouds
 * [2.0.x](/reference/javadoc/2.0.x/)
+* [1.9.x](/reference/javadoc/1.9.x/)
+* [1.8.x](/reference/javadoc/1.8.x/)
+* [1.7.x](/reference/javadoc/1.7.x/)
```
Nice!

nacx commented:
I've updated the PR. Now I've added a link to each list (there is no default one), and removed the links to the old archive site and the subscribe links. It is nicer to read the lists emails in PonyMail, and in each list there is a subscribe button at the bottom of the page. I think the site looks better with just the minimum links needed to direct users where they need to go.

If that does not get more subscribers (IMO) it is OK. We want to encourage participation in the lists, not to grow the number of subscribers.

Thoughts on this change?

demobox commented:
> Thoughts on this change?

Really like it! Two small-ish suggestions:

* Add something like "Ask questions about how to use jclouds here" to the description of the user list
* Optionally, add something like "Ask questions about how to contribute to jclouds here" to the description of the dev list
* Add a simple "subscribe" somewhere? Perhaps something like:

"The `user` list ([browse](#), [subscribe](#)) contains..."?

nacx commented:
>Add a simple "subscribe" somewhere? Perhaps something like:

I already tried locally keeping the subscribe links, but it was harder to read (at least for me) or to know where to click to get the information you are expecting to get. Too many links. I think providing just the two pointing to each list and let users find what they need is OK. If they want to subscribe, they'll find out how.

nacx commented:
>Optionally, add something like "Ask questions about how to contribute to jclouds here" to the description of the dev list

Not sure about this one. There are already the how to contribute links just below

jclouds-commentator commented:
Go to http://d5217a5dbcedee9468df-8db84ceb046a2260d615b30b9ea9ee0c.r28.cf5.rackcdn.com/ to review your changes.

demobox commented:
> If they want to subscribe, they'll find out how.

Based on the screenshot, I'm not sure where someone would look?

![screenshot](https://cloud.githubusercontent.com/assets/223702/21144513/fa654700-c118-11e6-91fb-7deb4433f5e1.png)

> There are already the how to contribute links just below

Just looking at the page, I think I would only be tempted to look at those if I wanted to _contribute_ code or docs. To me, there's nothing on the page now that encourages me to _participate_ rather than _contribute_ - no invitation to simply ask a question that I might have.

How about making "browse" and "participate" in `Browse and participate in our mailing lists` links?

nacx commented:
Hmmmm I agree. let's see how about this... :)

jclouds-commentator commented:
Go to http://54e8057648bd531d8fd0-2eb7f62ebbab77bd37361e9dba05a508.r32.cf5.rackcdn.com/ to review your changes.

demobox commented:
> Hmmmm I agree. let's see how about this... :)

+1, looks good to me! Thanks for the many updates, @nacx.

I see also that there's a "subscribe" link on the Pony Mail page, so anyone who _is_ interested should find it easy to get in touch:

![image](https://cloud.githubusercontent.com/assets/223702/21163423/f2320a94-c160-11e6-9ec3-8543aba71f97.png)

nacx commented:
There is! :)

>I've updated the PR. Now I've added a link to each list (there is no default one), and removed the links to the old archive site and the subscribe links. It is nicer to read the lists emails in PonyMail, and in each list there is a subscribe button at the bottom of the page

Thanks for the reviews @demobox! It definitely looks better now that the original version :)
Merging!

nacx commented:
Pushed to master as [c4f2455b](http://git-wip-us.apache.org/repos/asf/jclouds-site/commit/c4f2455b) and published! Thanks!
